### ISSUE TITLE
[1.21] Mixin issue - NeoForge

### ISSUE THREAD

**Issue by Collinvh** [Tslat/SmartBrainLib#64]
Log:
```WARN StatusConsoleListener The use of package scanning to locate plugins is deprecated and will be removed in a future release
WARN StatusConsoleListener The use of package scanning to locate plugins is deprecated and will be removed in a future release
WARN StatusConsoleListener The use of package scanning to locate plugins is deprecated and will be removed in a future release
WARN StatusConsoleListener The use of package scanning to locate plugins is deprecated and will be removed in a future release
[23:39:13.818] [main/INFO] [Launcher/MODLAUNCHER]: ModLauncher running: args [--launchTarget, forgeclientuserdev, --version, 21.0.6-beta, --assetIndex, asset-index, --assetsDir, C:\Users\Collin\.gradle\caches\minecraft\assets\1.21, --gameDir, ., --fml.neoForgeVersion, 21.0.6-beta, --fml.fmlVersion, 4.0.1, --fml.mcVersion, 1.21, --fml.neoFormVersion, 20240613.152323] 
[23:39:13.820] [main/INFO] [Launcher/MODLAUNCHER]: JVM identified as Amazon.com Inc. OpenJDK 64-Bit Server VM 21.0.3+9-LTS 
[23:39:13.822] [main/INFO] [Launcher/MODLAUNCHER]: ModLauncher 11.0.2+main.4612c65f starting: java version 21.0.3 by Amazon.com Inc.; OS Windows 10 arch amd64 version 10.0 
[23:39:13.919] [main/INFO] [loading.ImmediateWindowHandler/]: Loading ImmediateWindowProvider fmlearlywindow 
[23:39:13.993] [main/INFO] [EARLYDISPLAY/]: Trying GL version 4.6 
[23:39:14.182] [main/INFO] [EARLYDISPLAY/]: Requested GL version 4.6 got version 4.6 
[23:39:14.242] [main/INFO] [mixin/]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=union:/C:/Users/Collin/.gradle/caches/modules-2/files-2.1/net.fabricmc/sponge-mixin/0.13.4+mixin.0.8.5/8adf0cff1795a169141b9e16ad7b9f6e2527a990/sponge-mixin-0.13.4+mixin.0.8.5.jar%23136!/ Service=ModLauncher Env=CLIENT 
[23:39:14.429] [pool-2-thread-1/INFO] [EARLYDISPLAY/]: GL info: NVIDIA GeForce GTX 1070/PCIe/SSE2 GL version 4.6.0 NVIDIA 551.23, NVIDIA Corporation 
[23:39:14.492] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "neoforge-21.0.6-beta.jar" [locator: neoforge devenv provider ([C:\Users\Collin\Documents\GitHub\New-World\.gradle\repositories\ng_dummy_ng\net\neoforged\neoforge\21.0.6-beta\neoforge-21.0.6-beta.jar])] 
[23:39:14.501] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "neoforge-21.0.6-beta.jar" [locator: neoforge devenv provider ([C:\Users\Collin\Documents\GitHub\New-World\.gradle\repositories\ng_dummy_ng\net\neoforged\neoforge\21.0.6-beta\neoforge-21.0.6-beta.jar])] 
[23:39:14.509] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "main" [locator: userdev mods and services, reader: mod manifest] 
[23:39:14.517] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "neoforge-21.0.6-beta.jar" [locator: userdev mods and services, reader: mod manifest] 
[23:39:14.521] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "geckolib-neoforge-1.21-4.5.5.jar" [locator: userdev mods and services, reader: mod manifest] 
[23:39:14.529] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "TerraBlender-neoforge-1.21-4.0.0.1.jar" [locator: userdev mods and services, reader: mod manifest] 
[23:39:14.533] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found mod file "SmartBrainLib-neoforge-1.21-1.14.2.jar" [locator: userdev mods and services, reader: mod manifest] 
[23:39:14.555] [main/INFO] [loading.moddiscovery.ModDiscoverer/SCAN]: Found gamelibrary file "mixinextras-neoforge-0.3.5.jar" [locator: userdev mods and services, reader: mod manifest] 
[23:39:14.661] [main/WARN] [selection.JarSelector/]: Attempted to select a dependency jar for JarJar which was passed in as source: mixinextras-neoforge-0.3.5.jar. Using Mod File: C:\Users\Collin\.gradle\caches\modules-2\files-2.1\io.github.llamalad7\mixinextras-neoforge\0.3.5\ef3ffcf309dae656517b2f120189b1c9b21737dd\mixinextras-neoforge-0.3.5.jar 
[23:39:14.661] [main/INFO] [loading.moddiscovery.locators.JarInJarDependencyLocator/]: No dependencies to load found. Skipping! 
[23:39:14.811] [main/INFO] [loading.FMLServiceProvider/CORE]: Loading coremod script-engine for [neoforge] 
[23:39:15.556] [main/INFO] [mixin/]: Compatibility level set to JAVA_17 
[23:39:15.561] [main/INFO] [mixin/]: Compatibility level set to JAVA_21 
Exception in thread "main" org.spongepowered.asm.launch.MixinInitialisationError: Error initialising mixin config smartbrainlib.mixins.json
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.Config.create(Config.java:153)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:100)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:87)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformManager.addConfig(MixinPlatformManager.java:262)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformAgentDefault.prepare(MixinPlatformAgentDefault.java:46)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinContainer.prepare(MixinContainer.java:122)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformManager.createContainerFor(MixinPlatformManager.java:144)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformManager.addContainer(MixinPlatformManager.java:134)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformManager.addNestedContainers(MixinPlatformManager.java:152)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformManager.scanForContainers(MixinPlatformManager.java:213)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.platform.MixinPlatformManager.inject(MixinPlatformManager.java:186)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.MixinBootstrap.inject(MixinBootstrap.java:202)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.MixinLaunchPluginLegacy.initializeLaunch(MixinLaunchPluginLegacy.java:201)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.MixinLaunchPlugin.initializeLaunch(MixinLaunchPlugin.java:34)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.LaunchPluginHandler.lambda$announceLaunch$11(LaunchPluginHandler.java:100)
	at java.base/java.util.HashMap.forEach(HashMap.java:1429)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.LaunchPluginHandler.announceLaunch(LaunchPluginHandler.java:100)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:51)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.Launcher.run(Launcher.java:103)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.Launcher.main(Launcher.java:74)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210)
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69)
Caused by: java.lang.IllegalArgumentException: The specified resource 'smartbrainlib.mixins.json' was invalid or could not be read
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinConfig.create(MixinConfig.java:1290)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.Config.create(Config.java:148)
	... 24 more

> Task :runClient FAILED
32 actionable tasks: 1 executed, 31 up-to-date
```

Kinda looks like NeoForge expects a mixin?

**Comment by Collinvh** [comment]
It is defined in
Manifest.mf:
MixinConfigs: smartbrainlib.mixins.json

**Comment by Tslat** [comment]
Well damn
I'll push another version shortly

**Comment by Tslat** [comment]
Fixed in 1.14.3
